examples/move/reviews_rating/sources/service.move

```diff
@@ -0,0 +1,301 @@
+            reward: 1000000,
+            reward_pool: balance::zero(),
+            reviews: object_table::new<ID, Review>(ctx),
+            top_reviews: vector[],
+            overall_rate: 0,
+            name
+        };
+
+        let admin_cap = AdminCap {
+            id: object::new(ctx),
+            service_id
+        };
+
+        transfer::share_object(service);
+        transfer::public_transfer(admin_cap, tx_context::sender(ctx));
+        service_id
+    }
+
+    /// Writes a new review
+    public fun write_new_review(
+        service: &mut Service,
+        owner: address,
+        content: String,
+        overall_rate: u8,
+        clock: &Clock,
+        poe: ProofOfExperience,
+        ctx: &mut TxContext
+    ) {
+        assert!(poe.service_id == object::uid_to_inner(&service.id), EInvalidPermission);
+        let ProofOfExperience { id, service_id: _ } = poe;
+        object::delete(id);
+        let review = review::new_review(
+            owner,
+            object::uid_to_inner(&service.id),
+            content,
+            true,
+            overall_rate,
+            clock,
+            ctx
+        );
+        add_review(service, review, owner, overall_rate);
+    }
+
+    /// Writes a new review without proof of experience
+    public fun write_new_review_without_poe(
+        service: &mut Service,
+        owner: address,
+        content: String,
+        overall_rate: u8,
+        clock: &Clock,
+        ctx: &mut TxContext
+    ) {
+        let review = review::new_review(
+            owner,
+            object::uid_to_inner(&service.id),
+            content,
+            false,
+            overall_rate,
+            clock,
+            ctx
+        );
+        add_review(service, review, owner, overall_rate);
+    }
+
+    /// Adds a review to the service
+    fun add_review(
+        service: &mut Service,
+        review: Review,
+        owner: address,
+        overall_rate: u8
+    ) {
+        let id = review::get_id(&review);
+        let total_score = review::get_total_score(&review);
+        let time_issued = review::get_time_issued(&review);
+        object_table::add(&mut service.reviews, id, review);
+        update_top_reviews(service, id, total_score);
+        df::add<ID, ReviewRecord>(&mut service.id, id, ReviewRecord { owner, overall_rate, time_issued });
+        let overall_rate = (overall_rate as u64);
+        service.overall_rate = service.overall_rate + overall_rate;
+    }
+
+    /// Returns true if top_reviews should be updated given a total score
+    fun should_update_top_reviews(
+        service: &Service,
+        total_score: u64
+    ): bool {
+        let len = vector::length(&service.top_reviews);
+        if (len < MAX_REVIEWERS_TO_REWARD) {
+            return true
+        };
+        let review_id = vector::borrow(&service.top_reviews, len - 1);
+        if (total_score > get_total_score(service, *review_id)) {
+            return true
+        };
+        false
```

Comment:
```suggestion
        service.top_reviews.length() < MAX_REVIEWERS_TO_REWARD
          || total_score > service.get_total_score(service.top_reviews[len - 1])
```

Reply:
This might be a bit overkill to get it down to one expression, but the main comment here is to use receiver syntax.